Add support for use as pre-commit hook

proinsias opened the pull request:
* Adds configuration files needed for `image_optim` to be used as [pre-commit.com](https://pre-commit.com) hook.
* Also adds GitHub Action to regularly run [`pre-commit autoupdate`](https://pre-commit.com/#pre-commit-autoupdate) and create a pull request to update hook versions if needed.

toy commented on .github/workflows/actions.yml:
```diff
@@ -0,0 +1,24 @@
+name: GitHub Actions
+
+# yamllint disable-line rule:truthy
+on: [push, pull_request]
```
Are those good triggers?

toy commented on .github/workflows/actions.yml:
```diff
@@ -0,0 +1,24 @@
+---
+name: GitHub Actions
+
+# yamllint disable-line rule:truthy
```
As there is no yamllint configured to run, I think this comment shouldn't be here

toy commented on .github/workflows/actions.yml:
```diff
@@ -0,0 +1,24 @@
+          restore-keys: ${{ runner.os }}-gem-
+
+      - name: Pre-commit autoupdate
+        uses: browniebroke/pre-commit-autoupdate-action@v1.0.0
```
There seem to be a suggestion to create a PR after running this

toy reviewed (changes requested):
Thanks for contribution.
Unfortunately looks like the action [failed](https://github.com/proinsias/image_optim/actions/runs/733127746).
Could you also please describe what is the action supposed to do? Check if version is up to date with what pre-commit knows as the latest version?

proinsias commented:
@toy – thanks for the review. I believe I've addressed your comments. I realized I didn't need the GitHub Action as you're not using `pre-commit` for this repo itself.

toy commented:
Thanks for finishing the PR :)
Travis is failing only because failing check on up to date rubies.
Seemed to work when installing from your fork and specifying commit sha for rev, do I understand right that it would be better to create a release (tag) for it to work nicely?

toy commented:
Created v0.29.0 tag
